AES.md

```
@@ -0,0 +1,199 @@
+  - static (`.a`)
+  - dynamic (`.dylib`)
+This library format facilitates easy integration into various C++ projects, such as tools for analyzing game assets or general file system utilities, without requiring repeated compilation.
+
+The library's core functionality will be to provide a heuristic assessment of whether a given file is likely encrypted using a strong algorithm like AES. This analysis will be performed solely based on the file's byte-level statistical properties, without access to any decryption keys or prior knowledge of specific file formats. The goal is to offer a preliminary check: does the file content resemble random noise or structured data?
+
+## 2. Core Principle: Statistical Randomness
+
+The fundamental principle leverages the observation that data encrypted with robust algorithms (like AES) exhibits characteristics closely approximating a uniform random distribution of byte values. Each byte value (0-255) should occur with roughly equal probability, resulting in high *Shannon entropy*. This statistical profile contrasts sharply with typical unencrypted data.
+
+Consider these examples:
+*   **Text files** (`.txt`, `.cpp`, `.h`, etc.): Show strong biases towards common characters (e.g., space, 'e', 'a', 't' in English), specific line endings, and recurring keywords or syntax patterns.
```

markdownlint-cli2 (0.17.2)

Unordered list style
Expected: dash; Actual: asterisk

(MD004, ul-style)